hw/drivers/flash/enc_flash/include/enc_flash/enc_flash.h

```diff
@@ -31,6 +31,7 @@ extern "C" {
 struct enc_flash_dev {
     struct hal_flash efd_hal;
     const struct hal_flash *efd_hwdev; /* pointer to underlying hw dev */
+    struct os_sem ef_sem;
```

Comment:
Why was this locking added in this layer?

Reply:
i'm assuming this is being done because the AES engine can only support a single AES operation in progress at a time.  However, perhaps we need an AES driver that has this embedded inside of that.  We could have other users that are not the flash driver who may use the AES engine.

Reply:
Moving it the semaphore to eflash_da1469x_dev structure so that it's in the lowest layer. It will be accessed by the encryption driver for dialog